torch/_inductor/utils.py

```diff
@@ -1253,10 +1253,16 @@ def is_big_gpu(index_or_device: Union[int, torch.device] = 0) -> bool:
 
 
 @functools.lru_cache
-def get_num_sms() -> int:
+def get_max_num_sms() -> int:
     return torch.cuda.get_device_properties("cuda").multi_processor_count
 
 
+def get_num_sms() -> int:
+    """Handle experimental carveout if set otherwise return hardware SM count"""
+    carveout = torch._C._get_sm_carveout_experimental()
+    return carveout if carveout is not None else get_max_num_sms()
```

Comment:
could be wrong, but I thought these were the semantics:

```suggestion
    carveout = torch._C._get_sm_carveout_experimental()
    return get_max_num_sms() - (carveout if carveout is not None else 0)
```

Reply:
Ahh you are very right the formatting on this line was very weird for me and got confused: https://github.com/pytorch/pytorch/pull/144974/files#diff-74fcb26047c1df4024105d36ce22a36b77cf8cc93c28631d743e639b3d6066aeR413-R414